### PR TITLE
Fix crop_area_bytes_length calculation and add tests

### LINKED ISSUE #15386: Entanglement game panics while cropping image in canvas thread
http://entanglement.gopherwoodstudios.com/
```
index 29536 out of range for slice of length 28968 (thread CanvasThread, at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libcore/slice.rs:578)
stack backtrace:
   0:        0x10dff9351 - backtrace::backtrace::trace::h6545df4cf0bad269
   1:        0x10dff1fc4 - backtrace::capture::Backtrace::new::ha2ca403b809d6889
   2:        0x10cf9d3eb - servo::main::_$u7b$$u7b$closure$u7d$$u7d$::h85aa86e426463240
   3:        0x11356ad54 - std::panicking::rust_panic_with_hook::h33761bada49f3713
                        at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libstd/panicking.rs:556
   4:        0x11356ac34 - std::panicking::begin_panic<collections::string::String>
                        at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libstd/panicking.rs:517
   5:        0x11356ab52 - std::panicking::begin_panic_fmt::h4c0c43306d52c68f
                        at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libstd/panicking.rs:501
   6:        0x11356aab7 - rust_begin_unwind
                        at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libstd/panicking.rs:477
   7:        0x1135943a0 - core::panicking::panic_fmt::hbd633f652cd3a047
                        at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libcore/panicking.rs:69
   8:        0x113594499 - core::slice::slice_index_len_fail::h03f3b8ee39d37b51
                        at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libcore/slice.rs:578
   9:        0x10df5c937 - _$LT$core..ops..Range$LT$usize$GT$$u20$as$u20$core..slice..SliceIndex$LT$T$GT$$GT$::index::h04293639053b9ddf
  10:        0x10df4e0ae - core::slice::_$LT$impl$u20$core..ops..Index$LT$I$GT$$u20$for$u20$$u5b$T$u5d$$GT$::index::h3471b9a54a6b2422
  11:        0x10df04154 - _$LT$collections..vec..Vec$LT$T$GT$$u20$as$u20$core..ops..Index$LT$core..ops..Range$LT$usize$GT$$GT$$GT$::index::h1ad1eef08b37f28c
  12:        0x10df64cb6 - canvas::canvas_paint_thread::crop_image::h268bfdbd90d72e2b
  13:        0x10df6060d - canvas::canvas_paint_thread::CanvasPaintThread::draw_image::h4d0034b5b537d014
  14:        0x10df00ce2 - canvas::canvas_paint_thread::CanvasPaintThread::start::_$u7b$$u7b$closure$u7d$$u7d$::he49fa60cee0ae681
  15:        0x10deffe2a - _$LT$std..panic..AssertUnwindSafe$LT$F$GT$$u20$as$u20$core..ops..FnOnce$LT$$LP$$RP$$GT$$GT$::call_once::hdd81703e3d73574a
  16:        0x10df6e63e - std::panicking::try::do_call::hc8aa764ace046ad8
  17:        0x11356bc6a - __rust_maybe_catch_panic
                        at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/libpanic_unwind/lib.rs:98
  18:        0x10df6dd6a - std::panicking::try::h06d63f4ed6ae7124
  19:        0x10deef916 - std::panic::catch_unwind::h692c1a21a99f5971
  20:        0x10def117e - std::thread::Builder::spawn::_$u7b$$u7b$closure$u7d$$u7d$::h1ed031b0871efa30
  21:        0x10def8773 - _$LT$F$u20$as$u20$alloc..boxed..FnBox$LT$A$GT$$GT$::call_box::hfc0bf2c0bf5e3109
  22:        0x113569cd4 - std::sys::imp::thread::{{impl}}::new::thread_start
                        at /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/liballoc/boxed.rs:615
  23:     0x7fff91cdf898 - _pthread_body
  24:     0x7fff91cdf729 - _pthread_start
ERROR:servo: index 29536 out of range for slice of length 28968
```

### PR DESCRIPTION
**Pull request by n0max**
- Fix crop_area_bytes_length calculation (the height was multiplied by itself)
- Add tests for this error and for the case that the width is multiplied by itself

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #15386

- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15929)

### CODE REVIEW

File: tests/wpt/web-platform-tests/2dcontext/drawing-images-to-the-canvas/drawimage_html_image_14.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html class="reftest-wait">
+<meta charset="utf-8">
+<title></title>
```
Comment (nox) [inline tests/wpt/web-platform-tests/2dcontext/drawing-images-to-the-canvas/drawimage_html_image_14.html]: Add a title.

File: tests/wpt/web-platform-tests/2dcontext/drawing-images-to-the-canvas/drawimage_html_image_15.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html class="reftest-wait">
+<meta charset="utf-8">
+<title></title>
```
Comment (nox) [inline tests/wpt/web-platform-tests/2dcontext/drawing-images-to-the-canvas/drawimage_html_image_15.html]: Add a title.

**Review by nox — CHANGES_REQUESTED** [review]
Do not forget to update the manifest when you fix the tests.

**Comment by nox** [comment]
Seems good to go now, just needs a squash.

**Comment by nox** [comment]
```
diff --git a/tests/wpt/mozilla/meta/MANIFEST.json b/tests/wpt/mozilla/meta/MANIFEST.json
index b848a0645..366654cac 100644
--- a/tests/wpt/mozilla/meta/MANIFEST.json
+++ b/tests/wpt/mozilla/meta/MANIFEST.json
@@ -9583,11 +9583,6 @@
      {}
     ]
    ],
-   "mozilla/referrer-policy/generic/subresource/mozresource.pyc": [
-    [
-     {}
-    ]
-   ],
    "mozilla/referrer-policy/generic/subresource/stash.py": [
     [
      {}
@@ -25742,10 +25737,6 @@
    "333a80e3d4aec4a8be5566605fae5675a821085b",
    "support"
   ],
-  "mozilla/referrer-policy/generic/subresource/mozresource.pyc": [
-   "d4588e13a26732b38edb79fc11487a0542e87c40",
-   "support"
-  ],
   "mozilla/referrer-policy/generic/subresource/stash.py": [
    "97450e21d3280bb5950d3ad015d9b0041a5b9b2e",
    "support"
```

You need to update the manifest.

**Comment by jdm** [comment]
(That requires running `./mach update-manifest` and committing the changes).

**Review by nox — APPROVED** [review]

**Pull request merged**